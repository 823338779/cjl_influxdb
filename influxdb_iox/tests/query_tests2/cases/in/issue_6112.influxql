-- Query tests derived from https://github.com/influxdata/influxdb_iox/issues/6112
-- IOX_SETUP: InfluxQLSelectSupport

--
-- Single measurement queries
--

-- Projection wildcard, all tags and fields
-- IOX_COMPARE: sorted
SELECT * FROM m0;

-- Projection wildcard, only tags
-- IOX_COMPARE: sorted
SELECT *::tag, f64 FROM m0;

-- Projection wildcard, only fields
-- IOX_COMPARE: sorted
SELECT *::field FROM m0;

-- Projection regex, mixture of tags and fields
-- IOX_COMPARE: sorted
SELECT /64|tag0/ FROM m0;

-- Projection specific tags and fields
-- IOX_COMPARE: sorted
SELECT f64, tag0 FROM m0;

-- Explicitly select time column
-- IOX_COMPARE: sorted
SELECT f64, tag0, time FROM m0;

-- Validate some math functions
-- IOX_COMPARE: sorted
SELECT f64, floor(f64), ceil(f64) FROM m0;

-- Validate all scalar functions
-- -- IOX_COMPARE: sorted
-- TODO(sgc): log expects two arguments
-- TODO(sgc): asin and acos should cast NaN to NULL
-- SELECT f64, abs(f64), sin(f64), cos(f64), tan(f64),
--        asin(f64), acos(f64), atan(f64), atan2(f64, 1),
--        exp(f64), log(f64), ln(f64), log2(f64),
--        log10(f64), sqrt(f64), pow(f64, 2), floor(f64),
--        ceil(f64), round(f64)
-- FROM m0 LIMIT 1;

-- arithmetic operators
-- IOX_COMPARE: sorted
SELECT f64, f64 * 2, i64, i64 + i64 FROM m0;

-- bitwise operators
-- IOX_COMPARE: sorted
SELECT i64, i64 & 1 FROM m0;

-- Automatic type coercion integer â†’ float
-- IOX_COMPARE: sorted
SELECT f64 + i64 FROM m0;


-- Type cast postfix operator
-- -- IOX_COMPARE: sorted
-- TODO(sgc): cast operator not implemented
-- SELECT f64, f64::integer FROM m0;

-- Column alias behaviour
-- IOX_COMPARE: sorted
SELECT f64 AS f64_2, f64, f64, f64 FROM m0 LIMIT 1;

--
-- WHERE clause
--

-- Single tag
-- IOX_COMPARE: sorted
SELECT tag0, f64 FROM m0 WHERE tag0 = 'val00';

-- IOX_COMPARE: sorted
SELECT tag0, f64 FROM m0 WHERE tag0 =~ /^val0(1|2)/;

-- Conjunction (AND)
-- IOX_COMPARE: sorted
SELECT /tag(0|1)/, f64 FROM m0 WHERE tag0 = 'val00' AND tag1 = 'val10';

-- Disjunction (OR)
-- IOX_COMPARE: sorted
SELECT /tag(0|1)/, f64 FROM m0 WHERE tag0 = 'val00' OR tag1 = 'val10';

-- arithmetic
-- IOX_COMPARE: sorted
SELECT f64 FROM m0 WHERE f64 > 10 + 10;

-- bitwise
-- IOX_COMPARE: sorted
SELECT i64 FROM m0 WHERE i64 & 1 = 0;

-- time bounds

-- timestamp format %Y-%M-%D
-- IOX_COMPARE: sorted
SELECT i64 FROM m0 WHERE time > '2022-10-31';

-- timestamp format %Y-%M-%D %h:%m:%s
-- IOX_COMPARE: sorted
SELECT i64 FROM m0 WHERE time > '2022-10-31 02:00:10';

-- now() and duration
-- NOTE: 100000d is > 270 years, so this test should be ok for a while.
--       However, if this test is still in use in 270 years and it starts failing,
--       try increasing the number of days ðŸ˜‚
-- IOX_COMPARE: sorted
SELECT i64 FROM m0 WHERE time > now() - 100000d;

-- NOT NULL test
-- WHERE tag1 != '' is the equivalent to tag1 IS NOT NULL
-- TODO(sgc): This is working, but likely by accident
-- IOX_COMPARE: sorted
SELECT tag1, f64 FROM m0 WHERE tag1 != '';

-- NULL test
-- WHERE tag1 = '' is the equivalent to tag1 IS NULL
-- TODO(sgc): Not working, as expected
-- -- IOX_COMPARE: sorted
-- SELECT tag1, f64 FROM m0 WHERE tag1 = '';
