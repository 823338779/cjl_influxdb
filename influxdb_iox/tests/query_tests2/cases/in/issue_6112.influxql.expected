-- Test Setup: InfluxQLSelectSupport
-- InfluxQL: SELECT * FROM m0;
-- Results After Sorting
+----------------------+------+-----+-----+-------+-------+
| time                 | f64  | i64 | str | tag0  | tag1  |
+----------------------+------+-----+-----+-------+-------+
| 2022-10-31T02:00:00Z | 10.1 | 101 | hi  | val00 |       |
| 2022-10-31T02:00:00Z | 10.4 | 101 | lo  | val02 |       |
| 2022-10-31T02:00:00Z | 11.3 | 211 | lo  | val01 |       |
| 2022-10-31T02:00:10Z | 18.9 | 211 | lo  | val00 | val10 |
| 2022-10-31T02:00:10Z | 21.2 | 211 | hi  | val00 |       |
| 2022-10-31T02:00:20Z | 11.2 | 191 | lo  | val00 |       |
| 2022-10-31T02:00:30Z | 19.2 | 392 | lo  | val00 |       |
+----------------------+------+-----+-----+-------+-------+
-- InfluxQL: SELECT *::tag, f64 FROM m0;
-- Results After Sorting
+----------------------+-------+-------+------+
| time                 | tag0  | tag1  | f64  |
+----------------------+-------+-------+------+
| 2022-10-31T02:00:00Z | val00 |       | 10.1 |
| 2022-10-31T02:00:00Z | val01 |       | 11.3 |
| 2022-10-31T02:00:00Z | val02 |       | 10.4 |
| 2022-10-31T02:00:10Z | val00 | val10 | 18.9 |
| 2022-10-31T02:00:10Z | val00 |       | 21.2 |
| 2022-10-31T02:00:20Z | val00 |       | 11.2 |
| 2022-10-31T02:00:30Z | val00 |       | 19.2 |
+----------------------+-------+-------+------+
-- InfluxQL: SELECT *::field FROM m0;
-- Results After Sorting
+----------------------+------+-----+-----+
| time                 | f64  | i64 | str |
+----------------------+------+-----+-----+
| 2022-10-31T02:00:00Z | 10.1 | 101 | hi  |
| 2022-10-31T02:00:00Z | 10.4 | 101 | lo  |
| 2022-10-31T02:00:00Z | 11.3 | 211 | lo  |
| 2022-10-31T02:00:10Z | 18.9 | 211 | lo  |
| 2022-10-31T02:00:10Z | 21.2 | 211 | hi  |
| 2022-10-31T02:00:20Z | 11.2 | 191 | lo  |
| 2022-10-31T02:00:30Z | 19.2 | 392 | lo  |
+----------------------+------+-----+-----+
-- InfluxQL: SELECT /64|tag0/ FROM m0;
-- Results After Sorting
+----------------------+------+-----+-------+
| time                 | f64  | i64 | tag0  |
+----------------------+------+-----+-------+
| 2022-10-31T02:00:00Z | 10.1 | 101 | val00 |
| 2022-10-31T02:00:00Z | 10.4 | 101 | val02 |
| 2022-10-31T02:00:00Z | 11.3 | 211 | val01 |
| 2022-10-31T02:00:10Z | 18.9 | 211 | val00 |
| 2022-10-31T02:00:10Z | 21.2 | 211 | val00 |
| 2022-10-31T02:00:20Z | 11.2 | 191 | val00 |
| 2022-10-31T02:00:30Z | 19.2 | 392 | val00 |
+----------------------+------+-----+-------+
-- InfluxQL: SELECT f64, tag0 FROM m0;
-- Results After Sorting
+----------------------+------+-------+
| time                 | f64  | tag0  |
+----------------------+------+-------+
| 2022-10-31T02:00:00Z | 10.1 | val00 |
| 2022-10-31T02:00:00Z | 10.4 | val02 |
| 2022-10-31T02:00:00Z | 11.3 | val01 |
| 2022-10-31T02:00:10Z | 18.9 | val00 |
| 2022-10-31T02:00:10Z | 21.2 | val00 |
| 2022-10-31T02:00:20Z | 11.2 | val00 |
| 2022-10-31T02:00:30Z | 19.2 | val00 |
+----------------------+------+-------+
-- InfluxQL: SELECT f64, tag0, time FROM m0;
-- Results After Sorting
+------+-------+----------------------+
| f64  | tag0  | time                 |
+------+-------+----------------------+
| 10.1 | val00 | 2022-10-31T02:00:00Z |
| 10.4 | val02 | 2022-10-31T02:00:00Z |
| 11.2 | val00 | 2022-10-31T02:00:20Z |
| 11.3 | val01 | 2022-10-31T02:00:00Z |
| 18.9 | val00 | 2022-10-31T02:00:10Z |
| 19.2 | val00 | 2022-10-31T02:00:30Z |
| 21.2 | val00 | 2022-10-31T02:00:10Z |
+------+-------+----------------------+
-- InfluxQL: SELECT f64, floor(f64), ceil(f64) FROM m0;
-- Results After Sorting
+----------------------+------+-------+------+
| time                 | f64  | floor | ceil |
+----------------------+------+-------+------+
| 2022-10-31T02:00:00Z | 10.1 | 10    | 11   |
| 2022-10-31T02:00:00Z | 10.4 | 10    | 11   |
| 2022-10-31T02:00:00Z | 11.3 | 11    | 12   |
| 2022-10-31T02:00:10Z | 18.9 | 18    | 19   |
| 2022-10-31T02:00:10Z | 21.2 | 21    | 22   |
| 2022-10-31T02:00:20Z | 11.2 | 11    | 12   |
| 2022-10-31T02:00:30Z | 19.2 | 19    | 20   |
+----------------------+------+-------+------+
-- InfluxQL: SELECT f64, f64 * 2, i64, i64 + i64 FROM m0;
-- Results After Sorting
+----------------------+------+-------+-----+---------+
| time                 | f64  | f64_1 | i64 | i64_i64 |
+----------------------+------+-------+-----+---------+
| 2022-10-31T02:00:00Z | 10.1 | 20.2  | 101 | 202     |
| 2022-10-31T02:00:00Z | 10.4 | 20.8  | 101 | 202     |
| 2022-10-31T02:00:00Z | 11.3 | 22.6  | 211 | 422     |
| 2022-10-31T02:00:10Z | 18.9 | 37.8  | 211 | 422     |
| 2022-10-31T02:00:10Z | 21.2 | 42.4  | 211 | 422     |
| 2022-10-31T02:00:20Z | 11.2 | 22.4  | 191 | 382     |
| 2022-10-31T02:00:30Z | 19.2 | 38.4  | 392 | 784     |
+----------------------+------+-------+-----+---------+
-- InfluxQL: SELECT i64, i64 & 1 FROM m0;
-- Results After Sorting
+----------------------+-----+-------+
| time                 | i64 | i64_1 |
+----------------------+-----+-------+
| 2022-10-31T02:00:00Z | 101 | 1     |
| 2022-10-31T02:00:00Z | 101 | 1     |
| 2022-10-31T02:00:00Z | 211 | 1     |
| 2022-10-31T02:00:10Z | 211 | 1     |
| 2022-10-31T02:00:10Z | 211 | 1     |
| 2022-10-31T02:00:20Z | 191 | 1     |
| 2022-10-31T02:00:30Z | 392 | 0     |
+----------------------+-----+-------+
-- InfluxQL: SELECT f64 + i64 FROM m0;
-- Results After Sorting
+----------------------+---------+
| time                 | f64_i64 |
+----------------------+---------+
| 2022-10-31T02:00:00Z | 111.1   |
| 2022-10-31T02:00:00Z | 111.4   |
| 2022-10-31T02:00:00Z | 222.3   |
| 2022-10-31T02:00:10Z | 229.9   |
| 2022-10-31T02:00:10Z | 232.2   |
| 2022-10-31T02:00:20Z | 202.2   |
| 2022-10-31T02:00:30Z | 411.2   |
+----------------------+---------+
-- InfluxQL: SELECT f64 AS f64_2, f64, f64, f64 FROM m0 LIMIT 1;
-- Results After Sorting
+----------------------+-------+------+-------+-------+
| time                 | f64_2 | f64  | f64_1 | f64_3 |
+----------------------+-------+------+-------+-------+
| 2022-10-31T02:00:00Z | 10.1  | 10.1 | 10.1  | 10.1  |
+----------------------+-------+------+-------+-------+
-- InfluxQL: SELECT tag0, f64 FROM m0 WHERE tag0 = 'val00';
-- Results After Sorting
+----------------------+-------+------+
| time                 | tag0  | f64  |
+----------------------+-------+------+
| 2022-10-31T02:00:00Z | val00 | 10.1 |
| 2022-10-31T02:00:10Z | val00 | 18.9 |
| 2022-10-31T02:00:10Z | val00 | 21.2 |
| 2022-10-31T02:00:20Z | val00 | 11.2 |
| 2022-10-31T02:00:30Z | val00 | 19.2 |
+----------------------+-------+------+
-- InfluxQL: SELECT tag0, f64 FROM m0 WHERE tag0 =~ /^val0(1|2)/;
-- Results After Sorting
+----------------------+-------+------+
| time                 | tag0  | f64  |
+----------------------+-------+------+
| 2022-10-31T02:00:00Z | val01 | 11.3 |
| 2022-10-31T02:00:00Z | val02 | 10.4 |
+----------------------+-------+------+
-- InfluxQL: SELECT /tag(0|1)/, f64 FROM m0 WHERE tag0 = 'val00' AND tag1 = 'val10';
-- Results After Sorting
+----------------------+-------+-------+------+
| time                 | tag0  | tag1  | f64  |
+----------------------+-------+-------+------+
| 2022-10-31T02:00:10Z | val00 | val10 | 18.9 |
+----------------------+-------+-------+------+
-- InfluxQL: SELECT /tag(0|1)/, f64 FROM m0 WHERE tag0 = 'val00' OR tag1 = 'val10';
-- Results After Sorting
+----------------------+-------+-------+------+
| time                 | tag0  | tag1  | f64  |
+----------------------+-------+-------+------+
| 2022-10-31T02:00:00Z | val00 |       | 10.1 |
| 2022-10-31T02:00:10Z | val00 | val10 | 18.9 |
| 2022-10-31T02:00:10Z | val00 |       | 21.2 |
| 2022-10-31T02:00:20Z | val00 |       | 11.2 |
| 2022-10-31T02:00:30Z | val00 |       | 19.2 |
+----------------------+-------+-------+------+
-- InfluxQL: SELECT f64 FROM m0 WHERE f64 > 10 + 10;
-- Results After Sorting
+----------------------+------+
| time                 | f64  |
+----------------------+------+
| 2022-10-31T02:00:10Z | 21.2 |
+----------------------+------+
-- InfluxQL: SELECT i64 FROM m0 WHERE i64 & 1 = 0;
-- Results After Sorting
+----------------------+-----+
| time                 | i64 |
+----------------------+-----+
| 2022-10-31T02:00:30Z | 392 |
+----------------------+-----+
-- InfluxQL: SELECT i64 FROM m0 WHERE time > '2022-10-31';
-- Results After Sorting
+----------------------+-----+
| time                 | i64 |
+----------------------+-----+
| 2022-10-31T02:00:00Z | 101 |
| 2022-10-31T02:00:00Z | 101 |
| 2022-10-31T02:00:00Z | 211 |
| 2022-10-31T02:00:10Z | 211 |
| 2022-10-31T02:00:10Z | 211 |
| 2022-10-31T02:00:20Z | 191 |
| 2022-10-31T02:00:30Z | 392 |
+----------------------+-----+
-- InfluxQL: SELECT i64 FROM m0 WHERE time > '2022-10-31 02:00:10';
-- Results After Sorting
+----------------------+-----+
| time                 | i64 |
+----------------------+-----+
| 2022-10-31T02:00:20Z | 191 |
| 2022-10-31T02:00:30Z | 392 |
+----------------------+-----+
-- InfluxQL: SELECT i64 FROM m0 WHERE time > now() - 100000d;
-- Results After Sorting
+----------------------+-----+
| time                 | i64 |
+----------------------+-----+
| 2022-10-31T02:00:00Z | 101 |
| 2022-10-31T02:00:00Z | 101 |
| 2022-10-31T02:00:00Z | 211 |
| 2022-10-31T02:00:10Z | 211 |
| 2022-10-31T02:00:10Z | 211 |
| 2022-10-31T02:00:20Z | 191 |
| 2022-10-31T02:00:30Z | 392 |
+----------------------+-----+
-- InfluxQL: SELECT tag1, f64 FROM m0 WHERE tag1 != '';
-- Results After Sorting
+----------------------+-------+------+
| time                 | tag1  | f64  |
+----------------------+-------+------+
| 2022-10-31T02:00:10Z | val10 | 18.9 |
+----------------------+-------+------+